pyobis/nodes/nodes.py

```diff
@@ -2,82 +2,105 @@
 
-class NodesQuery(OBISQueryResult):
-    def __init__(self):
-        """
-        A NodesQuery object for fetching Nodes records.
-        """
 
-    def search(self, id=None, **kwargs):
-        """
-        Get OBIS nodes records
+def search(id=None, **kwargs):
+    """
+    Get OBIS nodes records
 
-        :param id: [String] Node UUID.
+    :param id: [String] Node UUID.
 
-        :return: A dictionary
+    :return: A NodesQuery Object
 
-        Usage::
+    Usage::
 
-            from pyobis.nodes import NodesQuery
-            nodes = NodesQuery()
-            nodes.search(id="4bf79a01-65a9-4db6-b37b-18434f26ddfc")
-        """
-        OBISQueryResult.url = obis_baseurl + "node/" + id
-        self.mapper = True
-        OBISQueryResult.args = {}
-        self.nodeid = id  # necessary to get mapper url
-        out = obis_GET(
-            OBISQueryResult.url,
-            OBISQueryResult.args,
-            "application/json; charset=utf-8",
-            **kwargs
-        )
+        from pyobis import nodes
+        data = nodes.search(id="4bf79a01-65a9-4db6-b37b-18434f26ddfc").execute()
+    """
+    url = obis_baseurl + "node/" + id
+    mapper = True
+    args = {}
 
-        return out
+    # return NodesQuery Object
+    return NodesResponse(url, args, mapper)
 
-    def activities(self, id=None, **kwargs):
-        """
-        Get OBIS nodes activities
 
-        :param id: [String] Node UUID.
+def activities(id=None, **kwargs):
+    """
+    Get OBIS nodes activities
 
-        :return: A dictionary
+    :param id: [String] Node UUID.
 
-        Usage::
+    :return: A NodesQuery object
 
-            from pyobis.nodes import NodesQuery
-            nodes = NodesQuery()
-            nodes.activities(id="4bf79a01-65a9-4db6-b37b-18434f26ddfc")
-        """
-        OBISQueryResult.url = obis_baseurl + "node/" + id + "/activities"
-        OBISQueryResult.args = {}
-        self.mapper = False
-        out = obis_GET(
-            OBISQueryResult.url,
-            OBISQueryResult.args,
-            "application/json; charset=utf-8",
-            **kwargs
-        )
-        return out
+    Usage::
 
-    def get_mapper_url(self):
-        """
-        Get the corresponding API URL for the query.
+        from pyobis import nodes
+
+        # build a query
+        query = nodes.activities(id="4bf79a01-65a9-4db6-b37b-18434f26ddfc")
+        query.execute() # execute the query to fetch the data
+        query.data # get fetched data
+        query.api_url # check the corresponding OBIS API URL for the query
+        query.mapper_url # check the corresponding OBIS Mapper URL for the query (if it exists)
+    """
+    url = obis_baseurl + "node/" + id + "/activities"
+    args = {}
+    mapper = False
+
+    # return a NodesQuery object
+    return NodesResponse(url, args, mapper)
 
-        :return: OBIS Mapper URL for the corresponding query
 
-        Usage::
+class NodesResponse:
```

Comment:
when instantiating a class without a complex parent, you should inherit from `object`, i.e.:

```
NodesResponse(object):
```

Reply:
Thank you so much for the review! I didn't know this fact, and I learned it today. I'll update the PR.